free-tunes/LogIn/LogInViewModel.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  LogInViewModel.swift
+//  Free-tunes
+//
+//  Created by Tshwarelo Mafaralala on 2022/03/24.
+//
+
+import Foundation
+
+protocol LogInViewModelDelegate {
+    func notify()
+    func show(error: String)
+}
+
+class LogInViewModel {
+    
+    private weak var delegate: LogInViewModelDelegate?
+    
+    init(delegate: LogInViewModelDelegate) {
+        self.delegate = delegate
+    }
+    
+// MARK: - Authentication
+    func login(username: String?, password: String?) -> Bool {
```

Comment:
These don't need to be optional 😄 `func login(username: String, password: String) -> Bool` use `?? ""` in your `ViewController` to pass in a default string if the user still hasn't entered anything.

Reply:
Noted😄